tcmur_cmd_handler.c

```diff
@@ -784,6 +810,7 @@ int tcmur_handle_writesame(struct tcmu_device *dev, struct tcmulib_cmd *cmd,
 			   tcmur_writesame_fn_t write_same_fn)
 {
 	struct tcmur_handler *rhandler = tcmu_get_runner_handler(dev);
+	struct tcmur_cmd_state *rcmd_state = tcmu_cmd_get_private(cmd);
```

Comment:
This is called from

handle_try_passthrough -> handle_cmd -> rbd/glfs -> tcmur_handle_writesame

and it looks like rcmd_state is not allocated yet. Below tcmur_handle_caw has the same call path, but you allocated in that function.